src/Plugin/TfaSetup/TfaHotpSetup.php

```diff
@@ -0,0 +1,188 @@
+<?php
+
+namespace Drupal\ga_login\Plugin\TfaSetup;
+
+use Base32\Base32;
+use Drupal\Core\Link;
+use Drupal\tfa\Plugin\TfaSetupInterface;
+use Drupal\tfa\Plugin\TfaValidation\TfaHotp;
+use Drupal\Core\Url;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\user\UserDataInterface;
+use Drupal\user\Entity\User;
+
+/**
+ * @TfaSetup(
+ *   id = "tfa_hotp_setup",
+ *   label = @Translation("TFA Hotp Setup"),
+ *   description = @Translation("TFA Hotp Setup Plugin"),
+ *   help_links = {
+ *    "Google Authenticator (Android/iPhone/BlackBerry)" = "https://support.google.com/accounts/answer/1066447?hl=en",
+ *    "Authy (Android/iPhone)" = "https://www.authy.com/thefuture#install-now",
+ *    "FreeOTP (Android)" = "https://play.google.com/store/apps/details?id=org.fedorahosted.freeotp",
+ *    "GAuth Authenticator (desktop)" = "https://github.com/gbraad/html5-google-authenticator"
+ *   }
+ * )
+ */
+class TfaHotpSetup extends TfaHotp implements TfaSetupInterface {
```

Comment:
Is the logic of this class ONLY what GA_Login needs?

Why wouldn't this be called GALoginHotpSetup?